src/app/pages/send/send-crypto-asset-form/family/stacks/components/stacks-recipient-field/components/recipient-bns-address-displayer.tsx

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react';
+import { FiCopy } from 'react-icons/fi';
+
+import { Box, Stack, color, useClipboard } from '@stacks/ui';
+import { SendCryptoAssetSelectors } from '@tests/selectors/send.selectors';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { AddressDisplayer } from '@app/components/address-displayer/address-displayer';
+import { Tooltip } from '@app/components/tooltip';
+
+interface BnsAddressDisplayerProps {
+  address: string;
+}
+export function RecipientBnsAddressDisplayer({ address }: BnsAddressDisplayerProps) {
+  const analytics = useAnalytics();
+  const { onCopy, hasCopied } = useClipboard(address);
+
+  const copyToClipboard = useCallback(() => {
+    void analytics.track('copy_recipient_bns_address_to_clipboard');
+    onCopy();
+  }, [analytics, onCopy]);
+
+  return (
+    <Stack alignItems="center" isInline mb="base">
+      <Box display="flex" flexWrap="wrap" fontSize={1} justifyContent="flex-start">
+        <AddressDisplayer address={address} />
+      </Box>
+      <Tooltip hideOnClick={false} label={hasCopied ? 'Copied!' : 'Copy address'} placement="right">
+        <Stack>
+          <Box
+            _hover={{ cursor: 'pointer' }}
+            as={FiCopy}
+            color={color('text-caption')}
+            data-testid={SendCryptoAssetSelectors.RecipientBnsAddressCopyToClipboard}
+            onClick={copyToClipboard}
+            size="16px"
+          />
+        </Stack>
```

Comment:
What does the stack do here?

Reply:
Tbh, I don't quite get it either. I removed it and the tooltip didn't work. Let me try again. I've seen this in several places.

Reply:
Was literally bc of `as={FiCopy}` here I think. Wrapped `FiCopy` in a div and made it clickable `as="button"` ...Stack is not needed. Will fix if I see other places.